File: app/src/main/java/org/ngbp/jsonrpc4jtestharness/rpc/xLink/model/XlinkResolution.java
package org.ngbp.jsonrpc4jtestharness.rpc.xLink.model;

public class XlinkResolution {
    public String xlink;
    public Disposition disposition;
    public Timing timing;

    public class Timing{
        public Double currentPosition;
        public String periodStart;
        public Double duration;
    }
    public class Disposition {
        public Integer code;
        public String description;
    }
}

File: app/src/main/java/org/ngbp/jsonrpc4jtestharness/rpc/xLink/model/XlinkResolution.kt
package org.ngbp.jsonrpc4jtestharness.rpc.xLink.model

data class XlinkResolution (
    var xlink: String? = null,
    var disposition: Disposition? = null,
    var timing: Timing? = null
)

data class Timing (
    var currentPosition: Double? = null,
    var periodStart: String? = null,
    var duration: Double? = null
)

data class Disposition (
    var code: Int? = null,
    var description: String? = null
)
